example/notify_inval_entry.c

```diff
@@ -260,6 +260,10 @@ static void* update_fs_loop(void *data) {
             if(options.only_expire) {
                 assert(fuse_lowlevel_notify_expire_entry
                    (se, FUSE_ROOT_ID, old_name, strlen(old_name), FUSE_LL_EXPIRE_ONLY) == 0);
```

Comment:
I'm not quite sure whether manually fiddling with `lookup_cnt` is the right thing to do.  Because as far as I can tell on older kernels this the flag `FUSE_LL_EXPIRE_ONLY` is silently ignored, and thus we get behaviour identical to the `else` branch and do get a forget request.

I'm not sure whether _not_ getting the forget request from the kernel is intentional?

Without this patch to reset `lookup_cnt`, we call `fuse_lowlevel_notify_expire_entry` even when the kernel has already forgotten about us, and thus we get an `-ENOENT` from the kernel instead of a `0`.

So an alternative fix for this example would be to change the assert to allow 0 or `-ENOENT`, and only in the latter case set `lookup_cnt` to `0`.

Keep in mind that this alternative fix would have us send one redundant `fuse_lowlevel_notify_expire_entry`.  (First the one that's actually useful, then the one that fails with `-ENOENT`.)